### ISSUE TITLE
review of tutorials/external_libs.rst

### ISSUE THREAD

**Issue by dogoepp** [resibots/limbo#127]
This is for easier discussion of pull request #115.

# `tutorials/external_libs.rst`
## around line 15
Why the default behavior is to return 1 ? Why do we not use ``res`` ?

Why does the library python script have to have #! /usr/bin/env python ?

## around line 241
Why return 1 both when we failed and succeeded ?

**Comment by dogoepp** [comment]
Answer from @costashatz:

> tutorials/external_libs.rst
> 
> around line 15
> 
> Why the default behavior is to return 1 ? Why do we not use res ?

There's no need to return `res`. Firstly, in the later versions of `waf`, `res` is not just a boolean anymore and secondly, `conf.fatal(msg)` prevents `waf` from continuing configuration...

> 
> Why does the library python script have to have #! /usr/bin/env python ?

It does not... I just put it always in my script to avoid using wrong version of python.. It's highly optional..

> 
> around line 69
> 
> Where is include_files defined ? Who has to fill it ? (I expect that it is the user, but it's not very clear in the tutorial)
> 

I have a comment just above that clearly (at least in my opinion) says that the user should specify a list of include files.

```py
# include_files is a list with the headers we expect to find
```

> around lines 68 and 96
> 
> Why do we define the res variable to True but never use it again ?

We do not need `res`. I forgot it there (copy paste). We should remove it..

> 
> around line 241
> 
> Why return 1 both when we failed and succeeded ?

Because we want to find `ROS` but it is not required. See above: it doesn't matter what we return, we have to use `conf.fatal(msg)` to prevent waf from continuing...

**Comment by dogoepp** [comment]
OK, all the `res` are removed in the file. Can we also clean the return statements that are not useful, as you claim ?

**Comment by jbmouret** [comment]
About the  #! /usr/bin/env python:
@costashatz <https://github.com/costashatz> : I am not sure this is used
when the script is loaded in another script (the version of python does not
change "on the fly")

@dogoepp: it is sometimes useful to give a hint to the source code editor
(for instance, most editors will not recognize wscripts as python script).

**Comment by costashatz** [comment]
> About the  #! /usr/bin/env python:
> @costashatz <https://github.com/costashatz> : I am not sure this is used
> when the script is loaded in another script (the version of python does not
> change "on the fly")

It is not, but I never said it is. I have it by default in my editor to put this line on the top of each python script..

**Comment by dogoepp** [comment]
gret, `#! /usr/bin/dev python` question solved.

**Comment by costashatz** [comment]
Fixed in fab634a. **Closing...**
